de.metas.material/dispo-service/src/main/java/de/metas/material/dispo/service/event/handler/pporder/PPOrderChangedHandler.java

```diff
@@ -87,20 +88,24 @@ public void handleEvent(@NonNull final PPOrderChangedEvent event)
 			updatedCandidatesToPersist.add(headerCandidate);
 		}
 
+		final ProductionDetail headerProductionDetail = ProductionDetail.cast(headerCandidate.getBusinessCaseDetail());
+		final DemandDetail headerDemandDetail = headerCandidate.getDemandDetail();
+
 		//
 		// Line candidates (demands, supplies)
 		if (event.isJustCompleted())
 		{
+			Loggables.addLog("PPOrder is just completed; create the demand and supply candidates for ppOrder lines");
+
 			PPOrderLineCandidatesCreateCommand.builder()
 					.candidateChangeService(candidateChangeService)
 					.candidateRepositoryRetrieval(candidateRepositoryRetrieval)
-					//
 					.ppOrder(event.getPpOrderAfterChanges())
-					// .supplyRequiredDescriptor(ppOrderEvent.getSupplyRequiredDescriptor())
+					.headerDemandDetail(headerDemandDetail)
 					.groupId(headerCandidate.getGroupId())
 					.headerCandidateSeqNo(headerCandidate.getSeqNo())
-					.advised(Flag.FALSE_DONT_UPDATE)
-					.pickDirectlyIfFeasible(Flag.FALSE_DONT_UPDATE)
+					.advised(headerProductionDetail.getAdvised())
+					.pickDirectlyIfFeasible(Flag.FALSE_DONT_UPDATE) // only the ppOrder's header supply product can be picked directly because only there might know the shipment schedule ID
```

Comment:
```suggestion
					.pickDirectlyIfFeasible(Flag.FALSE_DONT_UPDATE) // only the ppOrder's header supply product can be picked directly because only there we might know the shipment schedule ID
```